### DISCUSSION TITLE
Sonarr LXC running as root - permission issue

### DISCUSSION THREAD

**Discussion by chadnick84** [community-scripts/ProxmoxVE#354] [Q&A]
Running an unprivileged container of the most recent Sonarr LXC script. Sonarr doesn't appear to run as sonarr/media as has been suggested it should but as root. I've mounted an NFS share from the underlying host and mapped the group to a group id that root is a member of, can create and delete files when logged into the container itself but can't add the folder to Sonarr as it cites root doesn't have permission to write to it. Any ideas/suggestions?

**Comment by burgerga** [comment]
All arr apps run as root using the community scripts IIRC. What are your permissions on that folder on the proxmox host and in the sonarr LXC, and what is your id mapping?

**Comment by chadnick84** [comment]
Sorry I was basing that off the posts in this discussion - https://github.com/tteck/Proxmox/discussions/1738.

![Screenshot 2024-11-20 at 7 20 55 am](https://github.com/user-attachments/assets/a105205c-ec15-42ef-9032-f09d2c5abbd4)
Proxmox

![Screenshot 2024-11-20 at 7 21 10 am](https://github.com/user-attachments/assets/8f4e2dda-c55d-4f8e-a9b7-e751f6ffdb7a)
LXC Container

```
mp0: /mnt/lake/,mp=/lake
lxc.idmap: u 0 100000 1001
lxc.idmap: g 0 100000 1001
lxc.idmap: u 1001 1001 1
lxc.idmap: g 1001 1001 1
lxc.idmap: u 1002 101002 64534
lxc.idmap: g 1002 101002 64534
```

ID mapping

root is a member of group lxc_shares at both the proxmox and LXC container level.

Like I say, within the container as root I can write to the folders without issue.

**Reply by burgerga** [reply]
Hmm, nothing weird I can see. Did you check that other files in the container are also owned by 1001 and not by 101001? (https://forum.proxmox.com/threads/uid-mapping-and-permission-issue.126660/#post-553394)

And I have `chmod g+s` on the folders to make sure that newly created files are also owned by same group

**Comment by chadnick84** [comment]
The rest of the files in the container were owned by root. Changing the permissions to allow all users rwx sorted it, but wasn't my preference. In the end I ditched the id mapping, created new user and group on TrueNas with UID/GID of 100000 and changed the folder owner to those. Added my other users that should have access to those folders to that GID and that did the trick while not granting all users rwx. Thanks for your advice.
